### PR TITLE
Minor changes, don't merge

### PR DESCRIPTION
**Pull request by alyrow**
Merge pull request #22 from Minecraftian14/pre-master

I allow myself to correct my mistakes 😅 😅

### CODE REVIEW

**Comment by alyrow** [comment]
I think I have missed my change to merge...

**Comment by Minecraftian14** [comment]
Wow sooo many commits 😁

**Comment by Minecraftian14** [comment]
So it's like a pre-pre-master 🤣

**Comment by Minecraftian14** [comment]
Oh yeah, the last commit was "Experiment Successful: Stage 2"

**Comment by alyrow** [comment]
Time to review that 😨

File: src/main/java/com/alyrow/gdx/particle/Particle.java [hunk 1]
```diff
@@ -111,7 +111,7 @@ public void render(SpriteBatch batch, Camera camera, Array<PhysicForce> forces)
         sprite.setPosition(physicParticle.x, physicParticle.y);
         light.setPosition(physicParticle.x + physicParticle.width / 2, physicParticle.y + physicParticle.height / 2);
 
-        if(type == ParticleType.HALO && !sprite.getColor().equals(light.getColor())) {
+        if(type.isImage() && !sprite.getColor().equals(light.getColor())) {
```
Comment (alyrow) [inline src/main/java/com/alyrow/gdx/particle/Particle.java:114]: Need to modify that with ParticleType

File: src/main/java/com/alyrow/gdx/particle/ParticleSystem.java [hunk 2]
```diff
@@ -65,41 +68,46 @@
...
         this.world = world;
         this.camera = camera;
         batch = new SpriteBatch();
         time = TimeUtils.millis();
 
-        if(type == ParticleType.HALO) texture = new ParticleTexture("halo.png");
-        else if (type == ParticleType.NOTHING) texture = new ParticleTexture();
+//        if(type == ParticleType.HALO) texture = new ParticleTexture("halo.png");
+//        else if (type == ParticleType.NOTHING) texture = new ParticleTexture();
+        texture = this.type.getTexture();
```
Comment (alyrow) [inline src/main/java/com/alyrow/gdx/particle/ParticleSystem.java:100]: Need to do the breaking case

File: src/main/java/com/alyrow/gdx/particle/ParticleType.java [hunk 3]
```diff
@@ -1,11 +1,62 @@
...
+    CIRCLE("circle.png"),
+    DISCORD("discord.png"),
+    GRILL("grill.png"),
+    HALO("halo.png"),
+    SQUARE("square.png"),
+    STAR("star.png"),
+    TRIANGLE("triangle.png"),
+
+    TEXTURE(),
+    NOTHING();
```
Comment (alyrow) [inline src/main/java/com/alyrow/gdx/particle/ParticleType.java:20]: Modification to texture

File: src/main/java/com/alyrow/gdx/particle/ParticleType.java [hunk 4]
```diff
@@ -1,11 +1,62 @@
...
+        this.texture = new ParticleTexture(texture);
+    }
+
+    public void setTexture(String path) {
+        this.texture = new ParticleTexture(path);
+    }
+
+    public boolean isImage() {
+        return this != NOTHING && this != TEXTURE;
+    }
```
Comment (alyrow) [inline src/main/java/com/alyrow/gdx/particle/ParticleType.java:59]: Texture is image

File: src/main/java/com/alyrow/gdx/particle/physics/Fan.java [hunk 5]
```diff
@@ -1,26 +1,49 @@
...
-        return cache.scl(speed);
     }
 
+    @Override
+    public void read(Json json, JsonValue jsonData) {
+
+        center = new Vector2(jsonData.getFloat("center x"), jsonData.getFloat("center y"));
+        speed = jsonData.getFloat("speed");
+
+    }
```
Comment (alyrow) [inline src/main/java/com/alyrow/gdx/particle/physics/Fan.java:43]: Need to review in details the serialization

File: src/test/java/com/alyrow/gdx/particle/physics/MappedForceIOTest.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        // Reading
+        MappedForce force = TestRead();
+
+        return force;
+    }
+
+    private static void TestWrite() {
+
+        MappedForce force = new MappedForce(new Fan(0, 0, 20), true, null);
+        PixmapIO.writePNG(Gdx.files.absolute("D://i.png"), MappedForce.toPixmap(MappedForce.pack(force)));
```
Comment (alyrow) [inline src/test/java/com/alyrow/gdx/particle/physics/MappedForceIOTest.java:33]: Path

**Review by alyrow — COMMENTED** [review]
So all the comments are for me. I need to changes things, but it's such a progress

**Comment by alyrow** [comment]
I am merging it in another branch for some modifications

File: src/main/java/com/alyrow/gdx/particle/Particle.java (hunk 1 above)
Comment (alyrow) [reply src/main/java/com/alyrow/gdx/particle/Particle.java:114]: https://github.com/alyrow/box2d-light-particle/commit/821224c75cb5242b2da782d11c4eb7a360247870 ✔️

File: src/main/java/com/alyrow/gdx/particle/ParticleSystem.java (hunk 2 above)
Comment (alyrow) [reply src/main/java/com/alyrow/gdx/particle/ParticleSystem.java:100]: False alert
https://github.com/alyrow/box2d-light-particle/commit/a6a176eb1898c9fef3f38168ed68ed7699223ff7

File: src/main/java/com/alyrow/gdx/particle/ParticleType.java (hunk 3 above)
Comment (alyrow) [reply src/main/java/com/alyrow/gdx/particle/ParticleType.java:20]: Okay too
https://github.com/alyrow/box2d-light-particle/commit/a6a176eb1898c9fef3f38168ed68ed7699223ff7

File: src/main/java/com/alyrow/gdx/particle/ParticleType.java (hunk 4 above)
Comment (alyrow) [reply src/main/java/com/alyrow/gdx/particle/ParticleType.java:59]: https://github.com/alyrow/box2d-light-particle/commit/821224c75cb5242b2da782d11c4eb7a360247870 ✔️

File: src/test/java/com/alyrow/gdx/particle/physics/MappedForceIOTest.java (hunk 6 above)
Comment (alyrow) [reply src/test/java/com/alyrow/gdx/particle/physics/MappedForceIOTest.java:33]: https://github.com/alyrow/box2d-light-particle/commit/e72b205f6cdbd2825f1d9b0885a184492932ee62 ✔️

File: src/main/java/com/alyrow/gdx/particle/physics/Fan.java (hunk 5 above)
Comment (alyrow) [reply src/main/java/com/alyrow/gdx/particle/physics/Fan.java:43]: Such a great idea, when I will have finished optimisation, I will do that
https://github.com/alyrow/box2d-light-particle/commit/3529ee518b4d3a751cfdc9f7337f19b207e05fd2

**Comment by alyrow** [comment]
So I think I can now merge that in pre-master
https://github.com/alyrow/box2d-light-particle/pull/25

**Pull request merged**